sopel/modules/reddit.py

```diff
@@ -37,6 +40,7 @@
 
 
 domain = r'https?://(?:www\.|old\.|pay\.|ssl\.|[a-z]{2}\.)?reddit\.com'
+subreddit_url = r'%s/r?/([\w-]+)' % domain
```

Comment:
The `r` is not an optional part of the match. `reddit.com/subredditname` doesn't exist.

Also, be careful about leaving this pattern unterminated. Can't test right now, but I believe this pattern as written will match on submission links, too.

```suggestion
subreddit_url = r'%s/r/([\w-]+)/?$' % domain
```